gpu-operator/release-notes.rst

```diff
@@ -33,6 +33,85 @@ Refer to the :ref:`GPU Operator Component Matrix` for a list of software compone
 
 ----
 
+.. _v25.10.0:
+
+25.10.0
+=======
+
+New Features
+------------
+
+* Updated software component versions:
```

Comment:
We also updated:
- NVIDIA Kubernetes Device Plugin v0.18.0
- NVIDIA vGPU Device Manager v0.4.1
- NVIDIA Kubernetes Driver Manager v0.9.0
- NVIDIA MIG Manager v0.13.0